File: CS-Examples/03_Cells/GetCellDataType.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;

namespace GetCellDataType
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook.
            Workbook workbook = new Workbook();

            // Load the file from disk.
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Xls_2.xlsx");

            // Get the first worksheet.
            Worksheet sheet = workbook.Worksheets[0];

            //Get the cell types of the cells in range ¡°C13:F13¡±
            foreach (CellRange range in sheet.Range["H2:H7"])
            {
                XlsWorksheet.TRangeValueType cellType = sheet.GetCellType(range.Row, range.Column, false);
                sheet[range.Row, range.Column + 1].Text = cellType.ToString();
                sheet[range.Row, range.Column + 1].Style.Font.Color = Color.Red;
                sheet[range.Row, range.Column + 1].Style.Font.IsBold = true;
            }

            // Specify the filename for the resulting Excel file
            String result = "Result-GetCellDataType.xlsx";
            // Save the workbook to the specified file in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the MS Excel file.
            ExcelDocViewer(result);
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: CS-Examples/03_Cells/GetCellDataType.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet

Namespace GetCellDataType
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Instantiate a new Workbook object.
            Dim workbook As New Workbook()

            'Load the Excel file from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Template_Xls_2.xlsx")

            'Access the first worksheet in the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Iterate over each cell range in the specified range.
            For Each range As CellRange In sheet.Range("H2:H7")
                'Retrieve the type of the cell at the specified row and column.
                Dim cellType As XlsWorksheet.TRangeValueType = sheet.GetCellType(range.Row, range.Column, False)
                'Set the text value of the adjacent cell in the same row to the cell type.
                sheet(range.Row, range.Column + 1).Text = cellType.ToString()
                'Set the font color of the adjacent cell in the same row to red.
                sheet(range.Row, range.Column + 1).Style.Font.Color = Color.Red
                'Apply bold formatting to the font of the adjacent cell in the same row.
                sheet(range.Row, range.Column + 1).Style.Font.IsBold = True
            Next range
            'Specify the output file name.
            Dim result As String = "Result-GetCellDataType.xlsx"

            'Save the modified workbook to a file in Excel 2013 format.
            workbook.SaveToFile(result, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
